neutrons/SNAPRed
# Ewm3337/EWM3466 Refactor Normalization and Add dMin controls

## Description of work

This is the refactor of the Normalization Calibration process to enable its individual parts to be invoked individually, thus enabling certain parts of assessment to run much quicker.

## Explanation of work



## To test
Run through the Normalization Calibration flow, and when you get to the assessment step update the smoothing slider and or dmin.  Then hit recalculate.  Observe a speedy recalculation.

Change the grouping and recalculate.  This must update the whole data set.  Observe in regression it still works, albeit as slow as before.

Inputs used during development:
```
runNumber = "58810"
backgroundRunNumber = "58813"
samplePath = "/SNS/SNAP/shared/Calibration_dynamic/CalibrantSamples/Silicon_NIST_640D_001.json"
cifPath = "/SNS/SNAP/shared/Calibration/CalibrantSamples/Silicon_NIST_640d.cif"
groupPath = "/SNS/SNAP/shared/Calibration/Powder/PixelGroupingDefinitions/SNAPFocGrp_Column.lite.xml"
calibrationWorkspace = "/SNS/users/dzj/Desktop/58810_calibration_ws.nxs" # need to create this using diffc_test.py with run number 58810
smoothingParam = 0.50
calibrationStatePath = "/SNS/SNAP/shared/Calibration/Powder/04bd2c53f6bf6754/57514/v_1/CalibrationParameters.json" 
```
based on PR#164



## Link to EWM item


[EWM#3337](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=3337)
[EWM#3466](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=3466)
